File: app/src/main/java/ru/p3tr0vich/fuel/utils/UtilsFormat.java
package ru.p3tr0vich.fuel.utils;

import android.text.TextUtils;
import android.text.format.DateUtils;
import android.widget.EditText;

import java.text.DecimalFormat;

import ru.p3tr0vich.fuel.ApplicationFuel;

public class UtilsFormat {

    private static final DecimalFormat DECIMAL_FORMAT = new DecimalFormat("#.##");

    public static String dateTimeToString(long date, boolean withYear, boolean abbrevMonth) {
        int flags = DateUtils.FORMAT_SHOW_DATE;
        flags |= withYear ? DateUtils.FORMAT_SHOW_YEAR : DateUtils.FORMAT_NO_YEAR;
        if (abbrevMonth) flags |= DateUtils.FORMAT_ABBREV_MONTH;

//        flags |= DateUtils.FORMAT_SHOW_TIME;

        return DateUtils.formatDateTime(ApplicationFuel.getContext(), date, flags);
    }

    public static String dateToString(long date, boolean withYear) {
        return dateTimeToString(date, withYear, false);
    }

    public static float stringToFloat(String value) {
        if (TextUtils.isEmpty(value)) return 0;
        try {
            return Float.parseFloat(value);
        } catch (Exception e) {
            return 0;
        }
    }

    public static String floatToString(float value, boolean showZero) {
        final String strValue = DECIMAL_FORMAT.format(value).replace(',', '.');
        return showZero ? strValue : value == 0 ? "" : strValue;
    }

    public static String floatToString(float value) {
        return floatToString(value, true);
    }

    public static float editTextToFloat(EditText edit) {
        return stringToFloat(edit.getText().toString());
    }

    public static void floatToEditText(EditText edit, float value, boolean showZero) {
        edit.setText(floatToString(value, showZero));
    }
}

File: app/src/main/kotlin/ru/p3tr0vich/fuel/utils/UtilsFormat.kt
package ru.p3tr0vich.fuel.utils

import android.text.TextUtils
import android.text.format.DateUtils
import android.widget.EditText
import android.widget.TextView
import ru.p3tr0vich.fuel.ApplicationFuel
import java.text.DecimalFormat

object UtilsFormat {

    private val DECIMAL_FORMAT = DecimalFormat("#.##")

    @JvmStatic
    fun dateTimeToString(date: Long, withYear: Boolean, abbrevMonth: Boolean): String {
        var flags = DateUtils.FORMAT_SHOW_DATE

        flags = flags or if (withYear) DateUtils.FORMAT_SHOW_YEAR else DateUtils.FORMAT_NO_YEAR

        if (abbrevMonth) {
            flags = flags or DateUtils.FORMAT_ABBREV_MONTH
        }

        //        flags |= DateUtils.FORMAT_SHOW_TIME;

        return DateUtils.formatDateTime(ApplicationFuel.context, date, flags)
    }

    @JvmStatic
    fun dateToString(date: Long, withYear: Boolean): String {
        return dateTimeToString(date, withYear, false)
    }

    @JvmStatic
    fun stringToFloat(value: String): Float {
        if (TextUtils.isEmpty(value)) return 0f

        return try {
            java.lang.Float.parseFloat(value)
        } catch (e: Exception) {
            0f
        }
    }

    @JvmStatic
    @JvmOverloads
    fun floatToString(value: Float, showZero: Boolean = true): String {
        val strValue = DECIMAL_FORMAT.format(value).replace(',', '.')

        return when {
            showZero -> strValue
            value == 0f -> ""
            else -> strValue
        }
    }

    @JvmStatic
    fun editTextToFloat(edit: EditText?): Float {
        return stringToFloat(edit?.text.toString())
    }

    @JvmStatic
    fun floatToEditText(edit: EditText?, value: Float, showZero: Boolean) {
        edit?.setText(floatToString(value, showZero))
    }

    @JvmStatic
    fun floatToTextView(text: TextView?, value: Float, showZero: Boolean) {
        text?.text = floatToString(value, showZero)
    }
}